internal/cluster/statemachine.go

```diff
@@ -112,6 +111,16 @@ func NewClusterStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostRefreshCluster(statusInfoPendingForInput),
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
```

Comment:
I didn't understand why you separate the transition - can you explain?

Reply:
The rules that I changed are rules that the source state == destination state == ClusterStatusPendingForInput
If you pass the conditions to the rules, it is possible that these transitions won't change anything (possible, but not necessary).
On such cases it will get an error, since nothing have changed https://github.com/openshift/assisted-service/blob/master/internal/cluster/common.go#L71
So I added a separate transitions that handle cases where the source and the destination is ClusterStatusPendingForInput.
Just updates the validation info, if changed, without option to get an error if nothing have changed.